tests/test_client.py

```diff
@@ -110,6 +110,31 @@ def test_client_http_client_attributes(db, username, password):
     assert http_client.request_timeout == 80
     assert client.request_timeout == http_client.request_timeout
 
+    client_no_timeout = ArangoClient(
+        hosts="http://127.0.0.1:8529", request_timeout=None
+    )
+
+    assert client_no_timeout.request_timeout is None
+
+    client_no_timeout = ArangoClient(
+        hosts="http://127.0.0.1:8529",
+        http_client=DefaultHTTPClient(request_timeout=None),
+    )
+
+    assert client_no_timeout.request_timeout is None
+
+    class NoTimeoutHTTPClient(DefaultHTTPClient):
+        REQUEST_TIMEOUT = None
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(request_timeout=self.REQUEST_TIMEOUT, *args, **kwargs)
```

Comment:
previously, the `NoTimeoutHTTPClient` with a class `REQUEST_TIMEOUT` would work without needing an `__init__`. This is PR basically introduces the ability to support `REQUEST_TIMEOUT` again, but with the exception of needing an `__init__`